src/main.rs

```diff
@@ -869,10 +922,24 @@ fn bisect_nightlies(cfg: &Config, client: &Client) -> Result<BisectionResult, Er
                 } else {
                     last_failure = nightly_date;
                 }
+
+                // first week jump by two days
+                let jump_length = if current_jump < 7 {
+                    current_jump += 2;
+                    2
+                // from 2nd to 7th week jump weekly
+                } else if current_jump < 49 {
+                    current_jump += 7;
+                    7
+                // from 7th week jump by two weeks
+                } else {
+                    current_jump += 14;
```

Comment:
I am not sure about these `+=` here but if it works in practice seems fine.

It might be nice to extract a test here and a function that given a date gives an iterator of dates (or maybe just a vector with, say, 20 dates).